### PR TITLE
More Kubernetes monitor startup changes 

### PR DESCRIPTION
**Pull request by mheese**
`manager.New` is already reaching out to the Kubernetes API. We have to include a lot more in the infinite retry section.

Addresses https://github.com/aporeto-inc/aporeto/issues/2692

### CODE REVIEW

File: monitor/internal/pod/monitor.go [hunk 1]
```diff
@@ -196,47 +171,101 @@ func (m *PodMonitor) Resync(ctx context.Context) error {
...
-		close(z)
-		errCh <- ctx.Err()
-	}()
+	controllerStarted := make(chan struct{})
+
 	go func() {
-		if err := mgr.Start(z); err != nil {
-			errCh <- err
+		// manager.New already contacts the Kubernetes API
+		for {
```
Comment (abhijitherekar) [inline monitor/internal/pod/monitor.go:193]: THis loop which is present everywhere, is just make sure we get the manager and the controllers, else we
 are going keep on retrying, right?

File: monitor/internal/pod/monitor.go [hunk 2]
```diff
@@ -196,47 +171,101 @@ func (m *PodMonitor) Resync(ctx context.Context) error {
...
-		case <-time.After(5 * time.Second):
+		case <-ctx.Done():
+			close(z)
+		case <-time.After(warningMessageSleep):
 			// we give the controller 5 seconds to report back before we issue a warning
 			zap.L().Warn(startupWarningMessage)
 		case <-controllerStarted:
 			m.kubeClient = mgr.GetClient()
-			zap.L().Debug("pod: controller startup finished", zap.Duration("duration", time.Since(startTimestamp)))
+			zap.L().Error("pod: controller startup finished", zap.Duration("duration", time.Since(startTimestamp)))
```
Comment (abhijitherekar) [inline monitor/internal/pod/monitor.go]: Why is this a Error? The controllers started successfully if it comes here. 
Am I missing something?

Comment (abhijitherekar) [reply monitor/internal/pod/monitor.go]: ok, now I saw the runnable start function. Looks good, it should be Error. 
But I am wondering in what case can we come here. ???

File: monitor/internal/pod/monitor.go [hunk 3]
```diff
@@ -196,47 +171,101 @@ func (m *PodMonitor) Resync(ctx context.Context) error {
...
+			break
+		}
+
+		// starting the manager is a bit awkward:
+		// - it does not use contexts
+		// - we pass in a fake signal handler channel
+		// - we start another go routine which waits for the context to be cancelled
+		//   and closes that channel if that is the case
+
+		for {
```
Comment (abhijitherekar) [inline monitor/internal/pod/monitor.go:245]: So the monitor is going to keep on trying till the controllers start else the enforcer will fail to start. 
That means without the controllers we don't start the enforcer?

File: monitor/internal/pod/monitor.go (hunk 1 above)
Comment (mheese) [reply monitor/internal/pod/monitor.go:193]: exactly ... all errors must be handled here, that's why I also removed the error return

File: monitor/internal/pod/monitor.go (hunk 3 above)
Comment (mheese) [reply monitor/internal/pod/monitor.go:245]: yes

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "badf9cb6c7f195152c175a44b74b5002d0c069ae",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "a73306155e0dce6ee721c30daa09cc53ef124cd0",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "56ab0c276dd128ca3acbccbe65eec56b98d394f4",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "badf9cb6c7f195152c175a44b74b5002d0c069ae",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "15fb40fe7e60d6be4983719f533030837a8176d0",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "1c16faa3080bc3d2282667d514c1e81a05768543",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "3240c0eaffafacaee92f5ce0069425f5b00395a4",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "1c16faa3080bc3d2282667d514c1e81a05768543",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "1c16faa3080bc3d2282667d514c1e81a05768543",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "6fa96c79cd3250f590123f54d0c9e81e0449eae3",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "9e92ee364da57326b0909d693bab53d49d329a7e",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "1c16faa3080bc3d2282667d514c1e81a05768543",
    "pipeline": "master"
  }
]
```

**Comment by mheese** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "k8s-startup-2",
    "component": "enforcerd",
    "pr-id": "1607",
    "commit-sha": "cb475c29be84fdc139f909d82a71383754f8b5f8",
    "pipeline": "master"
  },
  {
    "project": "k8s-startup-2",
    "component": "trireme-lib",
    "pr-id": "984",
    "commit-sha": "1c16faa3080bc3d2282667d514c1e81a05768543",
    "pipeline": "master"
  }
]
```

**Review by abhijitherekar — APPROVED** [review]

**Pull request merged**